download_lean.py

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python3
+
+import tarfile, subprocess, sys
+
+lean_URL = "https://github.com/leanprover/lean/releases/download/v3.2.0/lean-3.2.0-linux.tar.gz"
```

Comment:
This needs to be generalised so that we don't use a hard-coded version number. Or perhaps changed if we want to compile the newest version of lean every time.